cloudify_types/cloudify_types/component/polling.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) 2017-2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from os import getenv
+
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+from cloudify_rest_client.exceptions import CloudifyClientError
+
+from .constants import POLLING_INTERVAL
+
+
+def any_bp_by_id(_client, _bp_id):
+    resource_type = 'blueprints'
+    return any_resource_by_id(_client, _bp_id, resource_type)
+
+
+def any_dep_by_id(_client, _dep_id):
+    resource_type = 'deployments'
+    return any_resource_by_id(_client, _dep_id, resource_type)
+
+
+def any_resource_by_id(_client, _resource_id, _resource_type):
+    return any(resource_by_id(_client, _resource_id, _resource_type))
+
+
+def all_deps_by_id(_client, _dep_id):
+    resource_type = 'deployments'
+    return all_resource_by_id(_client, _dep_id, resource_type)
+
+
+def all_resource_by_id(_client, _resource_id, _resource_type):
+    output = resource_by_id(_client, _resource_id, _resource_type)
+    if not output:
+        return False
+    return all(output)
+
+
+def resource_by_id(_client, _id, _type):
+    _resources_client = getattr(_client, _type)
+    try:
+        _resources = _resources_client.list(_include=['id'])
+    except CloudifyClientError as ex:
+        raise NonRecoverableError(
+            '{0} list failed {1}.'.format(_type, str(ex)))
+    else:
+        return [str(_r['id']) == _id for _r in _resources]
+
+
```

Comment:
From here and onward below, I have some ideas about how to improve them, but it will take a bunch of experimentation.

Reply:
I would love to hear what u thought of